drivers/usb/device/Kconfig

```diff
@@ -21,6 +21,30 @@ config USB_DW_IRQ_PRI
 	help
 	USB interrupt priority.
 
+config USB_DC_STM
+	bool
+	prompt "USB device controller driver for STM32 devices"
+	default n
+	help
+	STM32 family USB device controller Driver.
+
+config USB_DC_STM_EP_NUM
+	int
+	depends on USB_DC_STM
+	default 4
```

Comment:
Is it with or without control endpoint? Four is not much, STM32F401 has 3 IN and 3 OUT, maybe you can configure it in Kconfig.defconfig.series/Kconfig.defconfig.stm32f401xe? Or is there a method in the HAL to get the configuration for selected SoC?

Reply:
This is the minimum config I found and so the default one, STM32F401RE has 4 bidirectional endpoints (including control one). Maybe I'm wrong, but for now I only see STM32F4 datasheets describing only 4 endpoints for USB OTG FS peripheral mode. If not, then we can for sure overwrite the config in the appropriate defconfig.

Reply:
I wouldn't add the number of endpoints or the RAM size in the default usb device Kconfig as it is SOC specific. Could we just add it to SOC's Kconfig.defconfig.stm32fxxxxx?

Reply:
Need double confirmation, but since this is a USB_DC_STM32* config and it should be declared in the same Kconfig as the USB_DC_STM32 driver, this is for coherency and to avoid any build issue. This can be overwritten in specific defconfigs if necessary. I agree that this should/will go in the DT if we are unable to retrieve this info dynamically.

Reply:
They are USB_DC_STM32* configs but the user should not change them through the menuconfig as they are device specific. The SOC's RAM and FLASH size was declared like this before they were moved in the DT files. We could declare it in the Kconfig.defconfig.series file if they have the same value for all SOCs in the family.

Reply:
These props are hidden in menuconfig. I'm a bit torn, Actually these configs are related to the OTG FS device instance, not directly to the driver config. So they should be something like ST_STM32_USB_OTGFS_EP_NUM... Since @jfischer-phytec-iot and you suggest to move this in Kconfig.defconfig.series, I'm going to apply this change.

Reply:
Please don't rename it to ST_STM32_USB_OTGFS_EP_NUM. It will make it specific for SOCs with OTGFS and it have change when we add SOC's with USB 2.0 controller.

Reply:
Nop I just moved this in the stm32f4 series kconfig.

Reply:
Can we rename this USB_DC_STM32_NUM_EP